benchmarks/gbench/shp/gemm.cpp

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "../common/dr_bench.hpp"
+
+using T = float;
+
+static void Gemm_DR(benchmark::State &state) {
+  auto devices_ = dr::shp::get_numa_devices(sycl::default_selector_v);
+
+  std::size_t n_devices = devices_.size();
+
+  auto devices =
+      dr::shp::trim_devices(devices_, std::min(n_devices, devices_.size()));
+
+  dr::shp::init(devices);
+
+  std::size_t matrix_size = 10;
```

Comment:
Use m, n, k for dimensions even though we are likely to only do square matrices. It will be easier to change if we need to do experiments in the future. When you have it working, change to 32 * 1024 to match Ben's code.

Reply:
changing to 32 * 1024 causes an integer overflow in Stats, for now I changed it to 32